src/Layer/LabelLayer.js

```diff
@@ -0,0 +1,197 @@
+    }
+
+    // TODO: see how to factorize this behavior with FeatureProcessing which is
+    // similar.
+    update(context, layer, node, parent) {
+        if (!parent && node.children.length) {
+            // if node has been removed dispose three.js resource
+            ObjectRemovalHelper.removeChildrenAndCleanupRecursively(this, node);
+            return;
+        }
+
+        if (this.frozen || !node.visible) {
+            return;
+        }
+
+        if (node.layerUpdateState[this.id] === undefined) {
+            node.layerUpdateState[this.id] = new LayerUpdateState();
+        }
+
+        if (!node.layerUpdateState[this.id].canTryUpdate()) {
+            return;
+        }
+
+        const extentsDestination = node.getExtentsByProjection(this.source.projection) || [node.extent];
+
+        const extentsSource = [];
+        for (const extentDest of extentsDestination) {
+            const ext = this.source.projection == extentDest.crs ? extentDest : extentDest.as(this.source.projection);
+            if (extentInsideSource(ext, this.source)) {
+                node.layerUpdateState[this.id].noMoreUpdatePossible();
+                return;
+            }
+            extentsSource.push(extentDest);
+        }
+        node.layerUpdateState[this.id].newTry();
+
+        const command = {
+            layer: this,
+            extentsSource,
+            view: context.view,
+            threejsLayer: this.threejsLayer,
+            requester: node,
+        };
+
+        return context.scheduler.execute(command).then((result) => {
+            if (!result) { return; }
+
+            result.forEach((labels) => {
+                const labelsToAdd = [];
+
+                labels.forEach((label) => {
+                    if (label.parent || !node.extent.isPointInside(label.coordinates)) { return; }
+
+                    label.minlevel = node.extent.zoom;
+                    if (this.source.isVectorSource && label.needsMoveToLocal) {
+                        node.worldToLocal(label.position);
+                        label.needsMoveToLocal = false;
+                    }
+
+                    labelsToAdd.push(label);
+                });
+
+                if (!node.parent) {
+                    labelsToAdd.forEach(l => ObjectRemovalHelper.removeChildrenAndCleanupRecursively(this, l));
+                    return;
+                }
+
+                if (labelsToAdd.length > 0) {
+                    labelsToAdd.forEach((l) => {
+                        node.add(l);
+                        context.view.mainLoop.gfxEngine.label2dRenderer.domElement.appendChild(l.content);
+
+                        if (l.needsAltitude) {
+                            // TODO: add a Style property for the offset
+                            l.position.z = DEMUtils.getElevationValueAt(this.parent, l.coordinates) + 2;
+                        }
+
+                        l.visible = true;
+                        l.updateDimensions();
```

Comment:
not necessary?

Reply:
I have to do it after adding it to the DOM, I can't really see another place to do it, because I can get the dimensions only when the content of the Label is visible.